src/infra/ChatAnalyticsReporter.ts

```diff
@@ -0,0 +1,74 @@
+import type { EventPayload } from '../models/chat';
+import type ChatAnalyticsConfig from '../models/config/ChatAnalyticsConfig';
+
+interface EventsAPIResponse {
+  id: string;
```

Comment:
it's also possible for an error field to be returned in this response if its not a success.

`{
    "id": "01H3FNTWFVES5KF3J29V5V33PW",
    "errors": [...]
}`

Reply:
Are `errors` and `id` mutually exclusive? Or can there be both?

If there are `errors`, what will the status code be? Because right now we throw an error when if the status is not 2xx. 

We could potentially change this behavior so the user can access data about the error. Let me know what you think.

Reply:
ID is always returned as regardless if an event is valid or invalid we always record it. Errors will only be returned if the status is non 2xx.

Reply:
updated interface and error message to account for `errors` field